src/haddock/libs/libsubprocess.py

```diff
@@ -8,36 +8,25 @@
 from haddock.core.exceptions import CNSRunningError, JobRunningError
 
 
-class Job:
-    """A job to be executed by the engine."""
+class BaseJob:
+    """Base class for a subprocess job."""
 
-    def __init__(self, input, output, executable, *args, arg_first=False):
-        self.input = input
+    def __init__(self, input_, output, executable, *args):
+        self.input = input_
```

Comment:
here I changed to `input_` because `input` is a reserved python function, better not use it.